src/components/admin/storage-plans-tab.js

```diff
@@ -0,0 +1,162 @@
+
+function StoragePlansTab({ initialStoragePlans, onStoragePlansSaved }) {
+  const { uiLocale } = useLocale();
+  const { t } = useTranslation('storagePlansTab');
+  const storageApiClient = useSessionAwareApiClient(StorageApiClient);
+  const [storagePlans, setStoragePlans] = useState(cloneDeep(initialStoragePlans));
+  const [editedStoragePlan, setEditedStoragePlan] = useState(null);
+  const [storagePlanNamesInUse, setStoragePlanNamesInUse] = useState([]);
+  const [isStoragePlanModalVisible, setIsStoragePlanModalVisible] = useState(false);
+
+  const handleEditClick = storagePlan => {
+    const { _id, name, maxBytes } = storagePlan;
+    setEditedStoragePlan({ _id, name, maxBytes });
+    setStoragePlanNamesInUse(storagePlans.filter(plan => plan._id !== storagePlan._id).map(plan => plan.name));
+    setIsStoragePlanModalVisible(true);
+  };
+
+  const handleNewStoragePlanClick = () => {
+    setEditedStoragePlan(null);
+    setStoragePlanNamesInUse(storagePlans.map(plan => plan.name));
+    setIsStoragePlanModalVisible(true);
+  };
+
+  const handleDeleteClick = storagePlan => {
+    confirmStoragePlanDeletion(t, storagePlan, async () => {
+      try {
+        await storageApiClient.deleteStoragePlan(storagePlan._id);
+        const currentStoragePlans = await storageApiClient.getAllStoragePlans(true);
+        setStoragePlans(currentStoragePlans);
+        onStoragePlansSaved(currentStoragePlans);
+      } catch (error) {
+        handleApiError({ error, logger, t });
+        throw error;
+      }
+    });
+  };
+
+  const handleModalOk = async storagePlan => {
+    try {
+      if (storagePlan._id) {
+        await storageApiClient.updateStoragePlan({
+          storagePlanId: storagePlan._id,
+          name: storagePlan.name,
+          maxBytes: storagePlan.maxBytes
+        });
+      } else {
+        await storageApiClient.createStoragePlan({
+          name: storagePlan.name,
+          maxBytes: storagePlan.maxBytes
+        });
+      }
+
+      const currentStoragePlans = await storageApiClient.getAllStoragePlans(true);
+      setStoragePlans(currentStoragePlans);
+      onStoragePlansSaved(currentStoragePlans);
+      setEditedStoragePlan(null);
+      setStoragePlanNamesInUse([]);
+      setIsStoragePlanModalVisible(false);
+    } catch (error) {
+      handleApiError({ error, logger, t });
+    }
+  };
+
+  const handleModalCancel = () => {
+    setEditedStoragePlan(null);
+    setStoragePlanNamesInUse([]);
+    setIsStoragePlanModalVisible(false);
+  };
+
+  const renderStoragePlanActions = storagePlan => {
+    const actions = [
+      <a
+        key="edit"
+        onClick={() => handleEditClick(storagePlan)}
+        >
+        <EditIcon />
+      </a>
+    ];
+
+    if (!storagePlan.assignedUserCount) {
+      actions.push((
+        <a
+          key="delete"
+          onClick={() => handleDeleteClick(storagePlan)}
+          disabled={!!storagePlan.assignedUserCount}
+          >
```

Comment:
Unfortunately Antd hard-codes the color for these actions (including a blue-ish hover effect). Overriding it would require more overrides of ant styles. I initially tried (hence the superfluous `disabled` prop above, which I removed now), but then decided to leave it. Maybe I should have simply used a table here like we do on other pages. Should we change it (to table) maybe?

Reply:
Ok, I see. If you feel it's not a big effort and not overcomplicating the UI, then I'd vote for the table change. Up to you!

Reply:
Then I'll go for the table. The cards look nice but they'd introduce another type of UI we have nowhere else.

Reply:
Poifect!